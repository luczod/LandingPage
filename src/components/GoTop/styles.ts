import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    position: fixed;
    right: 3rem;
    bottom: 2rem;
    z-index: 6;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 4rem;
    width: 4rem;
    opacity: 0.8;
  `}
`;
