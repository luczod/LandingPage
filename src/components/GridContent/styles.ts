import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: 0;
    margin: 0 auto;
  `}
`;

export const HtmlWrap = styled.div`
  ${({ theme }) => css`
    margin: 0;
  `}
`;
