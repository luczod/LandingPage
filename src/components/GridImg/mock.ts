export default {
  bg: false,
  title: "My image grid",
  description:
    "Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde",
  img_grid: [
    {
      id: 8,
      image: {
        data: [
          {
            id: 8,
            attributes: {
              name: "Teste 1",
              alternativeText: "Teste 1",
              url: "https://source.unsplash.com/random/800x800?r=1",
            },
          },
        ],
      },
    },
    {
      id: 9,
      image: {
        data: [
          {
            id: 9,
            attributes: {
              name: "Teste 2",
              alternativeText: "Teste 2",
              url: "https://source.unsplash.com/random/800x800?r=2",
            },
          },
        ],
      },
    },
    {
      id: 10,
      image: {
        data: [
          {
            id: 10,
            attributes: {
              name: "Teste 3",
              alternativeText: "Teste 3",
              url: "https://source.unsplash.com/random/800x800?r=3",
            },
          },
        ],
      },
    },
  ],
};
