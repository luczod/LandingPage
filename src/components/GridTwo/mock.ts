export default {
  title: "Grid two columns",
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Perspiciatis harum odio tenetur consequatur obcaecati, earum perferendis commodi id repellat beatae, maiores numquam quos enim necessitatibus ipsam fugiat, incidunt consectetur aperiam!`,
  srcImg: "assets/images/javascript.svg",
  bg: true,
};
