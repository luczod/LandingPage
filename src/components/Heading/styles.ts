import styled, { DefaultTheme, css } from "styled-components";
type OnlyKeys = keyof typeof titleSize;
interface ISProps {
  $colorark: boolean;
  size: OnlyKeys;
  $uppertitle?: boolean;
}

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.md};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.lg};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xl};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhg};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xl};
  }
`;

export const Title = styled.h1<ISProps>`
  ${({ theme, size, $uppertitle, $colorark: colorDark }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    text-transform: ${$uppertitle ? "uppercase" : "none"};
  `}
`;
