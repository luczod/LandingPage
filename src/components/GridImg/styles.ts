import styled, { css } from "styled-components";
import { Container as TextComponent } from "../Textfn/styles";

interface IPros {
  open?: boolean;
}

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
  `}
`;

export const GridItem = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 1rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;

export const ModalImg = styled.div`
  ${({ theme }) => css`
    display: none; /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    padding-top: 100px; /* Location of the box */
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0 0 0 / 0.3); /* color-Black w/ opacity */
    border-radius: 1rem;

    & > span {
      position: fixed;
      cursor: pointer;
      top: 4rem;
      right: 2rem;
      color: ${theme.colors.secondaryColor};
      font-size: ${theme.font.sizes.hg};
      &::hover {
        color: ${theme.colors.white};
      }
      @media only screen and (max-width: 768px) {
        left: 1rem;
      }
    }

    & > img {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      border-radius: 1rem;
      /* 100% Image Width on Smaller Screens */
      @media only screen and (max-width: 768px) {
        width: 100%;
        padding: 0 0.8rem;
      }
    }
    & > div {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      text-align: center;
      font-size: ${theme.font.sizes.md};
      color: ${theme.colors.white};
      padding: 10px 0;
      height: 150px;
    }
  `}
`;
