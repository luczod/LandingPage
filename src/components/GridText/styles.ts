import styled, { css } from "styled-components";
import { Title as ItemTitle } from "../Heading/styles";
import { Container as TextComponent } from "../Textfn/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    overflow: hidden;
    width: 100%;
    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridItem = styled.article`
  ${({ theme }) => css`
    ${ItemTitle} {
      position: relative;
      left: 5rem;
    }

    ${ItemTitle}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhg};
      top: -2rem;
      left: -5rem;
      transform: rotate(8deg);
    }
  `}
`;
