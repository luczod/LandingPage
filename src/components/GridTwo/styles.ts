import styled, { DefaultTheme, css } from "styled-components";
import { Title } from "../Heading/styles";

interface IProps {
  background: boolean;
}

export const Container = styled.div<IProps>`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`;

export const TextColumn = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.lg};
      text-align: center;
    }
  `}
`;

export const ImgWrap = styled.div`
  ${({ theme }) => css``}
`;

export const Imgfn = styled.img`
  ${({ theme }) => css``}
`;
