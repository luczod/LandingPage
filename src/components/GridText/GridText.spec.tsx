import { renderTheme } from "../../styles/render-theme";
import { GridText } from ".";
import mock from "./mock";

describe("<GridText/>", () => {
  it("should render texts in a grid", () => {
    const { container } = renderTheme(<GridText {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
