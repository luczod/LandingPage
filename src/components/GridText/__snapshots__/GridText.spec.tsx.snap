// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridText/> > should render texts in a grid 1`] = `
.c4 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 3.2rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
  overflow: hidden;
  width: 100%;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 6.4rem;
  top: -2rem;
  left: -5rem;
  transform: rotate(8deg);
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  border-bottom: 0.2rem solid #DDDDDD;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 2.4rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3 c4"
        >
          My grid
        </h1>
        <div
          class="c5 c6"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c7"
        >
          <article
            class="c8"
          >
            <h1
              class="c3 c9"
            >
              Teste 1
            </h1>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </article>
          <article
            class="c8"
          >
            <h1
              class="c3 c9"
            >
              Teste 2
            </h1>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </article>
          <article
            class="c8"
          >
            <h1
              class="c3 c9"
            >
              Teste 3
            </h1>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </article>
        </div>
      </div>
    </div>
  </section>
</div>
`;
